### ISSUE TITLE
AI glitched out when seeking item diving suit + AI keeps taking leg injuries when not looking in Idle

### ISSUE THREAD

**Issue by NilanthAnimosus** [Regalis11/Barotrauma#1362]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
I think oxygen may have gone down causing the AI to seek a diving suit and not stop even once I returned the oxygen levels but what was interesting was what I saw, I don't have reproduction so all I have is two clips with some debugdraw:

[Clip A](https://i.gyazo.com/ac7b23bf78564786914e853126d4db44.mp4)
[Clip B](https://i.gyazo.com/91c3ab2f93c7b53667532025cb890348.mp4)

He hung around the room, seeking a diving suit presumably, I saw a diving suit floating and following his character (Visible near start of clip B and behind the character in clip A)

This diving suit followed the character. as much as I would say its a syncing issue this was actually single player and I was just seeking to have some junctions deteriorate.

It's as if he pulled it off the shelf, failed to wear it despite wearing no outer clothing, had it follow him and kept staying around the room trying to take it from a cabinet it didn't exist in anymore, unable to pick it up. Im not sure what the AI code could have done to take it out but not wear it (Doing this through a door so it doesn't let him interact but does take it off the shelf, then repeats with it no longer on the shelf? I don't know, see clips to be as confused as me).

on top of this, the AI when I'm not looking keeps managing to break one or both of his legs, I think find safety or idle is causing them to break their legs somehow (I don't know, jumping off ladders when oxygen gets low?) on the typhon.

**Version**
0.8.9.7 steam closed alpha - windows, singleplayer.

**Comment by itchyOwl** [comment]
That leg thing I've seen happen many times when a door gets slammed on the character. The main collider can be seen next to the door and the ragdoll is stuck for a while until it manages to replace itself correctly.

The AI is apparently trying to operate an item that it thinks requires a diving suit (a pump?). If it were due to lack of oxygen the bot would be in the "find safety state".

There is something wrong in the steering, since the bot tries to walk left but the path shows the next path is on the right, at the top of the ladders. They sometimes miss nodes, which should cause the bot to return though. Not sure why it tries to go left. It cannot open the door, because the button is not where the current node is. Not sure why the suit is floating there either. I've never encountered that before.

Did I understand correctly that the bot tried to take the suit through door? There was a bug that allowed bots to take items from containers through walls. I fixed it by adding a check that the character has to be in the same room (hull) than the container. So it shouldn't be possible now. But if it somehow can still interact with the container, that could explain the floating? Need to dig into the code...

**Comment by NilanthAnimosus** [comment]
Oh when I stated the leg thing as in leg injuries, I had noticed before this issue occurred my electrician had two 50-100+ damage wounds on his legs. more than he took severe internal damage injuries to his legs than getting stuck in a door to my amusement. don't think a door caused those. also why he was so slow in getting up that ladder and going anywhere.

you can tell from the electricians health in one of the clips for a moment how much damage hes done to his legs, he has no burns or any other injuries than internal damage leg injuries (2 large ones, one for each leg).

It'd be strange but my mechanic was operating reactor and my electrician on idle. so I think the operate target would have been the reactor, and the reactor room was getting low on oxygen...I don't recall any water anywhere but honestly that could have been me putting out fires and leaving the water command on (Which adds water to a hull) - but it probably was not enough for me to see if it was that.

As for the taking it through a door, I knew he went from the left side (not right side) to reach the suit, having gone from reactor - but I hadn't witnessed it myself, so I figure somehow perhaps he tried to do it through a door which I believe would fail any caninteract checks (Which I think the item pickup code uses) - but I imagine that could still dislodge it from its container if the code was that way, so it was more a suggestion of what could have occurred.

**Comment by itchyOwl** [comment]
If the reactor room lacked oxygen, the bot should first take diving gear before going there. It was first more natural (they wen't to look first), but then I changed it, because it felt that it just caused too much running around. So now they should know if they need the gear before going there. So that probably was the reason. But if the both ends up in a place with low oxygen, it should enter the find safety state. Currently there needs to more than 50% water or less than 50% oxygen for the bot to think that it needs the gear. But the room safety is calculated dynamically, so when finding safety, the actual level of oxygen and water is used as a factor (together with fires and enemies) to evaluate the safety of the room.

But let's hope that the issue got fixed.
